### ISSUE TITLE
`strict_equality` should report on constant `==` operations

### ISSUE THREAD

**Issue by KotlinIsland** [python/mypy#12082]
```py
def foo() -> bool:
    return None == None
```
`1 == 1` 
`True == False`

These codes show no warning about an always true check.

# mypy in real life 😳
```py
class Thing:
    thing: str | None
    def something(self, that: Thing) -> bool:
        if self.thing is not None or that.thing is not None:
            return True
        else:
            return self.thing == that.thing
```
Would be nice if mypy could report on this code.
